### ISSUE TITLE
Building a development environment is broken on Linux

### ISSUE THREAD

**Issue by jvkersch** [enthought/vpsearch#34]
CI runs (e.g. https://github.com/enthought/vpsearch/runs/8236514520?check_suite_focus=true) have started failing because the Python headers cannot be found. Looking at the compiler invocation, I see that the compiler looks in `/home/vagrant/workspace/Buildsystem/python/runtime/core/include/python3.8`, which looks like a directory on a build VM somewhere.

This is with EDM 3.2.5 but the problem persists with a more recent version of EDM. The Python runtime is 3.8.12+3  on rh7_x86_64.

```
gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -O2 -fPIC -Ivpsearch -I/tmp/pip-build-env-0xon16ad/overlay/lib/python3.8/site-packages/numpy/core/include -I/home/jvkersch/.edm/envs/vpsearch/include -I/home/vagrant/workspace/Buildsystem/python/runtime/core/include/python3.8 -c vpsearch/_vpsearch.cpp -o /tmp/tmpzcnox8ib.build-temp/vpsearch/_vpsearch.o -std=c++11
```

**Comment by jvkersch** [comment]
Needless to say, I don't have much insight into how this problem arose or how to work around it.

**Comment by rahulporuri** [comment]
Like I mentioned in https://github.com/enthought/vpsearch/pull/32#issuecomment-1240269711, this is an issue with the Python build itself and is being investigated internally.

**Comment by mdickinson** [comment]
A quick-fix solution is to restrict the version of `setuptools` used in `pyproject.toml`. Versions strictly older than setuptools 65.2 should work.

**Comment by jvkersch** [comment]
Thanks for the fix in #35, @mdickinson! This does fix the issue.

> It would probably make sense to remove the dependence on EDM from the continuous integration

I agree this would be the best course of action, and I will open an issue for that. We were tied to EDM for a very long time because there was an EDM-specific egg for the Parasail dependency, containing both the headers and shared object files needed to build the Cython extension. Recently, an [ambitious young developer](https://github.com/jeffdaily/parasail-python/pull/62) patched up the [parasail-python wheel](https://pypi.org/project/parasail/) available via PyPI, and in theory we could now rely on that dependency instead.

This would change the build aspects for vpsearch internally (an EDM egg would have to be built for the parasail-python dependency) so it is not without consequences. I will reach out via email to assess whether this can be done.
